airbyte-server/src/test/java/io/airbyte/server/RequestLoggerTest.java

```diff
@@ -35,95 +42,194 @@
+    private static final int SUCCESS_CODE = 200;
+    private static final String errorPrefix = RequestLogger
+        .createLogPrefix(REMOTE_ADDR, METHOD, ERROR_CODE, URL)
+        .toString();
+    private static final String successPrefix = RequestLogger
+        .createLogPrefix(REMOTE_ADDR, METHOD, SUCCESS_CODE, URL)
+        .toString();
+
+    static Stream<Arguments> logScenarios() {
+      return Stream.of(
+          Arguments.of(INVALID_JSON_OBJECT, NON_ACCEPTED_CONTENT_TYPE, ERROR_CODE, errorPrefix),
+          Arguments.of(INVALID_JSON_OBJECT, ACCEPTED_CONTENT_TYPE, ERROR_CODE, errorPrefix),
+          Arguments.of(VALID_JSON_OBJECT, NON_ACCEPTED_CONTENT_TYPE, ERROR_CODE, errorPrefix),
+          Arguments.of(VALID_JSON_OBJECT, ACCEPTED_CONTENT_TYPE, ERROR_CODE, errorPrefix + " - " + VALID_JSON_OBJECT),
+          Arguments.of(INVALID_JSON_OBJECT, NON_ACCEPTED_CONTENT_TYPE, SUCCESS_CODE, successPrefix),
+          Arguments.of(INVALID_JSON_OBJECT, ACCEPTED_CONTENT_TYPE, SUCCESS_CODE, successPrefix),
+          Arguments.of(VALID_JSON_OBJECT, NON_ACCEPTED_CONTENT_TYPE, SUCCESS_CODE, successPrefix),
+          Arguments.of(VALID_JSON_OBJECT, ACCEPTED_CONTENT_TYPE, SUCCESS_CODE, successPrefix + " - " + VALID_JSON_OBJECT));
+    }
+
+    @Mock
+    private ContainerRequestContext mRequestContext;
+    @Mock
+    private ContainerResponseContext mResponseContext;
+
+    private RequestLogger requestLogger;
+
+    @ParameterizedTest
+    @MethodSource("logScenarios")
+    @DisplayName("Check that the proper log is produced based on the scenario")
+    public void test(final String requestBody, final String contentType, final int status, final String expectedLog) throws IOException {
+      // We have to instanciate the logger here, because the MDC config has been changed to log in a
+      // temporary file.
+      requestLogger = new RequestLogger(MDC.getCopyOfContextMap(), mServletRequest);
+
+      stubRequestContext(mRequestContext, requestBody);
+
+      Mockito.when(mResponseContext.getStatus())
+          .thenReturn(status);
+
+      Mockito.when(mServletRequest.getHeader("Content-Type"))
+          .thenReturn(contentType);
+
+      // This is call to set the requestBody variable in the RequestLogger
+      requestLogger.filter(mRequestContext);
+      requestLogger.filter(mRequestContext, mResponseContext);
+
+      final String expectedLogLevel = status == SUCCESS_CODE ? "INFO" : "ERROR";
+
+      final String logs = IOs.readFile(logPath);
+      final Stream<String> matchingLines = logs.lines()
+          .filter(line -> line.endsWith(expectedLog))
+          .filter(line -> line.contains(expectedLogLevel));
+
+      Assertions.assertThat(matchingLines).hasSize(1);
+    }
+  }
 
-    Mockito.when(mServletRequest.getHeader("Content-Type"))
-        .thenReturn(contentType);
+  @Nested
+  @DisplayName("Logs correct requestBody")
+  public class RequestLoggerCorrectRequestBody {
+
+    /**
```

Comment:
how long does it take to run? should this be a "slow" test? @benmoriceau has context on what that means :D

Reply:
If this test takes a little bit to run, you can use the annotation @IntegrationTest or @SlowIntegration test. That will make the test to not run during the build command but it will still be running on the CI.

Reply:
It only takes a second to run, as that's the maximum sleep time that each thread can take (and they all run in parallel)